docs/glossary.md

```diff
@@ -0,0 +1,123 @@
+
+## CFC (Contextual Flow Control)
+
+A security model combining information flow control with contextual integrity; enforces policies on how data is used, attached to schemas and validated both statically and dynamically.
+
+## Charm
+
+Charm is a [spell] invocation binding set of [cell]s as inputs and set of [cell]s as outputs, creating an execution graph. It may help to think of [spell] as an open electric circuit, in this case [charm] would be a closed electric circuit as current will flow through it. Different analogy could be to think of [charm] as a process, where's [spell] would be a program and [cell]s would be program inputs and outputs.
+
+## CRDT (Conflict-free Replicated Data Type)
+
+A data structure that can resolve conflicts automatically in distributed systems. Used selectively, e.g. for collaborative text editing.
+
+## Space
+
+Space is primarily a sharing boundary, designed to enforce access control. Spaces are identified by unique [did:key] identifiers.
+
+> ℹ️ Currently each space has a corresponding sqlite database to store all of its state.
+
+Space can be queried and updated using [memory protocol], which describes state in terms of [fact](#fact)s.
+
+## Fact
+
+Is a record of state in time represented using `{ the, of, is, cause }` tuples. E.g consider following fact: _The_ **color** _of_ **sky** _is_ **blue** it would directly translated to
+`{ the: "color", of: "object:sky", is: "blue" }`.
+
+> ℹ️ The `cause` filed is used to establis causal references, it effectively represents a logical time per fact as opposed to global time.
+
+In practice we use `the` field to describe kind of the information value (`is` field) is provided about subject entity (`of` field). Predominantly `the` is `"application/json"` as we store [cell] contents as JSON values and consequently `is` field is a JSON value [cell]s hold at discrete points in time.
+
+The `of` field is a unique identifier represented via URI. In practice it usually a [merkle-reference] derived from some seed data with `of:` scheme prefix.
+
+## Memory
+
+Memory is an abstraction over [space] and an information system adhering to [The Value of Values] design principles. Abstraction provides efficient way to access current state - current facts about various entities, while still providing a way to recall facts that had being succeeded by the new ones.
+
+Memory also provides interface for accreting new information through an interface with [compare and swap (CAS)][CAS] semantics.
+
+> ℹ️ Please note that layers above [memory] do not follow same principals or operate at the level of [fact]s, instead they use more traditional document-oriented semantics and reference state by the address inside the mutable memory space.
+
+
+## Deno
+
+A JavaScript/TypeScript runtime used on the server side of Open Ocean.
```

Comment:
perhaps we mention it is also used as our build system (perhaps @jsantell can help with the exact words here)